### PR TITLE
Generate leaf setters

### PR DESCRIPTION
**Pull request by kjahed**
This PR adds the '-generate_leaf_setters' flag which enables the generation of leaf setters (SetXXX). The generated methods assign their parameter (or its address) to the corresponding fields depending on the fields' data type.

### CODE REVIEW

**Review by wenovus — CHANGES_REQUESTED** [review]
What you have LGTM, could you also set this flag in a few places where `GenerateLeafGetters:     true` is set in https://github.com/openconfig/ygot/blob/master/gogen/codegen_test.go? to make sure that the generated code compiles and that we have good coverage across the different YANG types.

**Review by wenovus — APPROVED** [review]
LGTM, thanks! @robshakir do you want to take a quick look before we merge this? There was a similar review you did in the past if you want to reload some context: https://github.com/openconfig/ygot/pull/281

File: gogen/gogen.go
Comment (robshakir) [inline gogen/gogen.go:1685]:
```suggestion
	
	if goOpts.GenerateLeafSetters {
```

File: gogen/gogen.go
Comment (robshakir) [inline gogen/gogen.go:1689]:
```suggestion
	}

```

File: gogen/gogen_test.go
Comment (robshakir) [inline gogen/gogen_test.go:1934]: Not blocking in this PR, but I wonder whether in the future we should support `SetLeafUnion<Type>` which would also hide needing to create the wrapper type. This would be similar to the way that the proto library does `oneof` fields.

**Review by robshakir — APPROVED** [review]

**Comment by robshakir** [comment]
Thanks @wenovus - this LGTM. Comments are minor here, but please accept the suggestions just for code style reasons :-)

File: gogen/gogen_test.go
Comment (wenovus) [reply gogen/gogen_test.go:1934]: Nice idea, unfortunately we can't easily do this right now because Go generics doesn't support methods whose parameters are generic because it complicates the interface system: https://go.dev/doc/faq#generic_methods

**Pull request merged**
